main.go

```diff
@@ -341,7 +341,7 @@ func Main(config *Config, configTest bool, buildVersion string, logger *logrus.L
 	//handshakeMACKey := config.GetString("handshake_mac.key", "")
 	//handshakeAcceptedMACKeys := config.GetStringSlice("handshake_mac.accepted_keys", []string{})
 
-	serveDns := config.GetBool("lighthouse.serve_dns", false)
+	serveDns := config.GetBool("lighthouse.am_lighthouse", false) && config.GetBool("lighthouse.serve_dns", false)
```

Comment:
I think a warning log in the event `serve_dns` is true while `am_lighthouse` is false would be helpful

Reply:
done!